wls-gui-wahllokalsystem/tests/composables/common/logging.spec.ts

```diff
@@ -0,0 +1,214 @@
+import { afterEach, describe, expect, it, vi } from "vitest";
+
+import { useLogging } from "@/composables/common/logging.ts";
+
+describe("logging.ts", () => {
+  let unitUnderTest: ReturnType<typeof useLogging>;
+
+  const loggerName = "loggerName";
+
+  afterEach(() => {
+    vi.resetAllMocks();
+    vi.clearAllMocks();
+  });
+
+  describe("log", () => {
+    it("should_log_when_loggingIsActive", () => {
+      allowLogging(true);
+      unitUnderTest = useLogging(loggerName);
+
+      const consoleLogMock = vi
+        .spyOn(console, "log")
+        .mockImplementation(() => undefined);
+
+      const logMessage = "logMessage";
+      unitUnderTest.log(logMessage);
+
+      expect(consoleLogMock).toHaveBeenCalledWith(
+        `${loggerName}: ${logMessage}`
+      );
+
+      consoleLogMock.mockReset();
+    });
+
+    it.each([undefined, new Error("dummy error"), "a string"])(
+      "should_log_when_loggingIsActiveWithAdditionalData'%s'",
+      (additionalData) => {
+        allowLogging(true);
+        unitUnderTest = useLogging(loggerName);
+
+        const consoleLogMock = vi
+          .spyOn(console, "log")
+          .mockImplementation(() => undefined);
+
+        const logMessage = "logMessage";
+        unitUnderTest.log(logMessage, additionalData);
+
+        expect(consoleLogMock).toHaveBeenCalledWith(
+          `${loggerName}: ${logMessage}`,
+          additionalData
+        );
+
+        consoleLogMock.mockReset();
+      }
+    );
+
+    it("should_notLog_when_loggingIsActive", () => {
```

Comment:
bitte auch in den anderen describe blöcken anpassen
```suggestion
    it("should_notLog_when_loggingIsNotActive", () => {
```

Reply:
korrigiert mit f631d85b55c4e5151da0ed9403147fb69f1b8bd1